storybookjs/storybook
# UI: Expand sidebar for selected story when using composition

Closes #22878



## What I did

- Update the `isExpanded` variable when a `selectedStoryId` is found in the `index` provided to the `Ref`.
- The `Refs` component is expanded when the selected story is inside the Ref and either:
  - storybook is initialized
  - the user switches to a new story
- This follows the current behaviour for automatically expanding nodes in the `Tree` component

### Expanding when opening storybook:

![expand-on-refresh](https://github.com/storybookjs/storybook/assets/16689477/ef4038dc-5022-4d48-bd41-b52252b0aac3)

### Expand when switching story using shortcut:

![ref-expand-on-next-story](https://github.com/storybookjs/storybook/assets/16689477/e46c6437-9204-43ae-9ace-0798cd19e3c2)

## How to test

1. Create two sandboxes
2. Add the following to main.ts on the first storybook:
```
  refs: {
    other: {
      title: 'other',
      url: 'http://localhost:6007',
      expanded: false
    }
  }
```
3. Run the sandboxes with `yarn storybook` and `yarn storybook --port 6007`
4. Access a story within the composition and refresh the page to see the ref open automatically
4. Navigate between stories within the composition and to see the ref open automatically after being closed

## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers


`["cleanup", "BREAKING CHANGE", "feature request", "bug", "build", "documentation", "maintenance", "dependencies", "other"]`



### 🦋 Canary release



This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_
